core/domain/draft_upgrade_services.py

```diff
@@ -0,0 +1,85 @@
+# coding: utf-8
+#
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Commands that can be used to upgrade draft to newer Exploration versions."""
+import logging
+
+from core.domain import exp_domain
+from core.platform import models
+import utils
+
+(exp_models, feedback_models, user_models) = models.Registry.import_models([
+    models.NAMES.exploration, models.NAMES.feedback, models.NAMES.user
+])
+
+
+def try_upgrading_draft_to_exp_version(
+        draft_change_list, current_draft_version, to_exp_version, exp_id):
+    """Try upgrading a list of ExplorationChange domain objects to match the
+        latest exploration version.
+
+    For now, this handles only the scenario where all commits between
+    current_draft_version and to_exp_version are that migrate the state schema.
+
+    Args:
+        draft_change_list: list(ExplorationChange). The list of
+            ExplorationChange domain objects to upgrade.
+        current_draft_version: int. Current draft version.
+        to_exp_version: int. Target exploration version.
+        exp_id: str. Exploration id.
+
+    Returns:
+        list(ExplorationChange) or None. A list of ExplorationChange domain
+        objects after upgrade or None if upgrade fails.
+
+    Raises:
+        InvalidInputException.
```

Comment:
State when this happens (see other examples in codebase).

Reply:
Done